syntax = "proto3";

package dis;

message Status {
  bool sucess = 1;
  string message = 2;
}

message Empty {

}

message Address {
  int32 port = 1;
  string ip = 2;
}

message AddressMap {
  map<int32, Address> address_map = 1;
}

message Matrix {
  repeated float data = 1;
  int32 rows = 2;
  int32 cols = 3;
  string name = 4;
}

message MatrixRemoteSynceInfo {
  string name = 1;
  int32 row_beginl = 2;
  int32 row_endl = 3;
  int32 col_beginl = 4;
  int32 col_endl = 5;
}

service DisService {
  rpc GetAddressMap (Empty) returns (AddressMap);
  rpc SendAddress (Address) returns (Status);
  
  rpc GetMatrixFromRemote (MatrixRemoteSynceInfo) returns (Matrix);
};
